lib/segment/src/index/field_index/geo_index/mod.rs

```diff
@@ -204,7 +204,9 @@ impl GeoMapIndex {
             return CardinalityEstimation::exact(0);
         }
 
-        let common_hash = common_hash_prefix(values);
+        let Some(common_hash) = common_hash_prefix(values) else {
```

Comment:
I propose to check directly here if `values` are empty. For `common_hash_prefix` return error is `values` are empty

Reply:
all other logic is sound if values are empty